### PR TITLE
Document Koji integration more fully

### PR DESCRIPTION
**Pull request by twaugh**
Added some metadata-level documentation.

### CODE REVIEW

File: docs/koji.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+## Koji integration
+
+The `koji` pre-build plugin and the `koji_promote` exit plugin provide integration with [Koji](https://docs.pagure.org/koji/).
```
Comment (lcarva) [inline docs/koji.md]: `add_filesystem` does too

File: docs/koji.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Pre-build plugin
+
+The `koji` pre-build plugin creates a yum repofile to allow instructions in the Dockerfile to have access to RPMs available from the appropriate Koji tag.
+
+## Exit plugin
```
Comment (lcarva) [inline docs/koji.md]: Do you think it's worth listing out a somewhat full json which includes all the attributes listed in this doc? Might be easier to visualize it all.

File: docs/koji.md (hunk 1 above)
Comment (twaugh) [inline docs/koji.md]: Mentioned this.

File: docs/koji.md (hunk 2 above)
Comment (twaugh) [inline docs/koji.md]: Yes, added.

**Comment by twaugh** [comment]
Updated.

File: docs/koji.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          "rootfs": {...},
+          "config": {...},
+          ...
+        },
+        "tags": [
+          "latest",
+          "1.0.0-1",
+          "1.0.0"
+        ]
+      }
```
Comment (lcarva) [inline docs/koji.md]: Missing comma.

File: docs/koji.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+          "arch": "x86_64"
+        }
+      },
+      "docker": {
+        "id": "sha256:abc123def...",
+        "parent_id": "sha256:123def456...",
+        "repositories": [
+          "registry.example.com/product/package-name:1.0.0-1",
+          "registry.example.com/product/package-name@sha256:123def..."
+        ]
```
Comment (lcarva) [inline docs/koji.md]: Missing comma.

**Comment by twaugh** [comment]
Updated.

File: README.md [hunk 5]
```diff
@@ -13,7 +13,8 @@ Python library with command line interface for building docker images.
  * build inside a docker container (so your builds are separated between each other)
  * git as a source to your Dockerfile (you may specify commit/branch and path to Dockerfile within the git repo)
  * collect build logs
- * integration with [koji](http://koji.fedoraproject.org/koji/) build system
+ * integration with
+   [koji](https://github.com/projectatomic/atomic-reactor/blob/master/docs/koji.md) build system
```
Comment (lcarva) [inline README.md]: It'd be nice to have this link being relative, so it behaves properly on forked projects: `../blob/master/docs/koji.md`
https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

Although, it'd be even nicer if it was relative to the current branch, but not sure if that's supported.

Comment (twaugh) [inline README.md]: Decided not to do this -- the relative links are pretty awkward-looking, and I'm not sure that they make any sense at all outside of the context of viewing the file on GitHub. Leaving them as full URIs allows the document to be read offline.

File: docs/koji.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+Each build creates a single output archive, in the [Combined Image JSON + Filesystem Changeset format](https://github.com/docker/docker/blob/master/image/spec/v1.2.md#combined-image-json--filesystem-changeset-format).
+
+# Type-specific build metadata
+
+For atomic-reactor container image builds the 'image' type is used, and so type-specific information is placed into the build.extra.image map. Note that this does not use build.extra.typeinfo.image -- the 'typeinfo' element was introduced after the 'image' type started to be used.
+
+Data which is placed here includes:
+
+- build.extra.image.autorebuild (boolean): true if this build was triggered automatically; false otherwise
```
Comment (vrutkovs) [inline docs/koji.md]: Nitpick: perhaps those should be wrapped in backticks: `build.extra.image.autorebuild`?

**Comment by twaugh** [comment]
Updated.

**Pull request merged**